synapse/cortex.py

```diff
@@ -4496,6 +4496,84 @@
         await self.feedBeholder('view:add', pack, gates=[iden])
         return pack
 
+    async def delViewWithLayer(self, iden):
+        '''
+        Delete a Cortex View and its write Layer by iden.
+
+        Note:
+            Any children of the View will have their parent view updated to
+            the deleted View's parent (if present). The deleted layer will also
+            be removed from any Views which contain it in their layer stack.
+        '''
+        view = self.views.get(iden)
+        if view is None:
+            raise s_exc.NoSuchView(mesg=f'No such view {iden=}', iden=iden)
+
+        if view.info.get('protected'):
+            mesg = f'Cannot delete view ({iden}) that has protected set.'
+            raise s_exc.CantDelView(mesg=mesg)
+
+        layriden = view.layers[0].iden
```

Comment:
minor nit, but this is unused

Reply:
ah, yeah i added it and forgot to use it, updated